config/config.go

```diff
@@ -41,6 +43,38 @@ func Config() *ConfigStruct {
 	return configData
 }
 
+//setMapValueEnvOverride recursively iterate over the keys and updates the map value accordingly
+func setMapValueEnvOverride(keys []string, mapValue map[string]interface{}, value interface{}) map[string]interface{} {
+	if len(keys) > 1  {
+		innerMapType := fmt.Sprintf("%T", mapValue[keys[0]])
```

Comment:
Using a type assertion with the optional 'ok' return value would be more straight forward here: https://go.dev/tour/methods/15

e.g. 
innerMapType, ok := mapValue[keys[0]]
if !ok {
  log.Warn("...")
  return mapValue
}

Reply:
that would just return true if the map contains that key.  It doesn't really say anything about even the data type of the innerMapType.